freerdp/3.1.0: Add version stream for freerdp 3.x

superseb opened the pull request:
Fixes: https://github.com/wolfi-dev/os/pull/9854 https://github.com/wolfi-dev/os/pull/10344

Related:
- Added new packages required for `cmake` to succeed
- `BUILTIN_CHANNELS` was removed in https://github.com/FreeRDP/FreeRDP/pull/7647
- Replaced tab characters found in original YAML (see below)

```
cat -A freerdp.yaml
...
  - runs: |$
      CFLAGS="$CFLAGS -fPIC" \$
      CXXFLAGS="$CXXFLAGS -fPIC" \$
      cmake -B build -G Ninja \$
      ^I-DCMAKE_BUILD_TYPE=MinSizeRel \$
      ^I-DCMAKE_INSTALL_PREFIX=/usr \$
      ^I-DCMAKE_INSTALL_LIBDIR=lib \$
      ^I-DWITH_ALSA=ON \$
      ^I-DWITH_CUPS=ON \$
      ^I-DWITH_CHANNELS=ON \$
      ^I-DBUILTIN_CHANNELS=OFF \$
      ^I-DWITH_DIRECTFB=OFF \$
      ^I-DWITH_FFMPEG=OFF \$
      ^I-DWITH_GSM=ON \$
      ^I-DWITH_GSTREAMER_1_0=ON \$
      ^I-DWITH_IPP=OFF \$
      ^I-DWITH_JPEG=ON \$
      ^I-DWITH_OPENSSL=ON \$
      ^I-DWITH_PCSC=OFF \$
      ^I-DWITH_PULSE=OFF \$
      ^I-DWITH_WAYLAND=ON \$
      ^I-DWITH_SERVER=ON \$
      ^I-DWITH_X11=ON \$
      ^I-DWITH_XCURSOR=ON \$
      ^I-DWITH_XEXT=ON \$
      ^I-DWITH_XKBFILE=ON \$
      ^I-DWITH_XI=ON \$
      ^I-DWITH_XINERAMA=ON \$
      ^I-DWITH_XRENDER=ON \$
      ^I-DWITH_XV=ON \$
      ^I-DWITH_ZLIB=ON \$
      ^I-DWITH_NEON=OFF$
       cmake --build build$
...
```

### Pre-review Checklist

#### For version bump PRs
<!-- remove if unrelated -->
- [ ] The `epoch` field is reset to 0

superseb committed:
Add version stream for freerdp 3.x

ajayk commented:
Thanks @superseb , freerdp supports multiple major versions is it possible to convert this to a version stream ,you can find an example [here](https://github.com/wolfi-dev/os/blob/main/argo-cd-2.9.yaml) of a version stream

superseb commented:
@ajayk Thanks, I changed the PR. I kept the original `freerdp.yaml` and added `freerdp-3.yaml` based on other files in this repository. Let me know if it needs to be different.

Also replaced some tab characters with spaces, apparently it worked but spaces will have more guarantee that it won't break.

ajayk commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,118 @@
+package:
+  name: freerdp
```
```suggestion
  name: freerdp-3
```

ajayk commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,118 @@
+  epoch: 0
+  description: FreeRDP client
+  copyright:
+    - license: Apache-2.0
```
```suggestion
    - license: Apache-2.0
  dependencies:
    provides:
      - freerdp=${{package.full-version}}
```

debasishbsws commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,121 @@
+package:
+  name: freerdp-3
+  version: 3.0.0
```
```suggestion
  version: 3.1.0
```
There is a new version available.

debasishbsws commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,121 @@
+  - uses: fetch
+    with:
+      uri: https://github.com/FreeRDP/FreeRDP/archive/${{package.version}}.tar.gz
+      expected-sha512: 6c9061674716ca8c83a3913222db4002d893d751b0072a8af10013e09462a9cc847689dc874e30c499ae0d5be73c464f610057744c771fcd678bc43185d0f923
```
```suggestion
      expected-sha512: 0a463c241d09ea354fdd943c010a2ecca1ad20f6ba5ea037422884f3b668ac6c4c30f73d12077368cd54f74418ef65b267628934a050fcbb3ff1e96d84ae954e
```

ajayk commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,121 @@
+  - uses: strip
+
+subpackages:
+  - name: freerdp-doc
```
```suggestion
  - name: ${{package.name}}-doc
```

ajayk commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,121 @@
+    pipeline:
+      - uses: split/manpages
+
+  - name: freerdp-dev
```
```suggestion
  - name: ${{package.name}}-dev
```

ajayk commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,121 @@
+        - freerdp
+    description: freerdp dev
+
+  - name: freerdp-libs
```
```suggestion
  - name: ${{package.name}}-libs
```

ajayk commented on freerdp-3.yaml:
```diff
@@ -0,0 +1,121 @@
+      - uses: split/dev
+    dependencies:
+      runtime:
+        - freerdp
```
```suggestion
        - freerdp-3
```

ajayk commented:
> @ajayk Thanks, I changed the PR. I kept the original `freerdp.yaml` and added `freerdp-3.yaml` based on other files in this repository. Let me know if it needs to be different.
> 
> Also replaced some tab characters with spaces, apparently it worked but spaces will have more guarantee that it won't break.

The last change is to adapt  sub package to reflect the package names , then we can get this merged , Thanks @superseb for your contribution again !!
